source/fab/source_tree.py

```diff
@@ -22,36 +22,56 @@ def visit(self, candidate: Path):
 
 
 class ExtensionVisitor(TreeVisitor):
-    def __init__(self, extension_map: Mapping[str, Analyser]):
+    def __init__(self,
+                 extension_map: Mapping[str, Union[Type[Task], Type[Command]]],
+                 command_flags_map: Mapping[Type[Command], List[str]],
+                 state: SqliteStateDatabase, workspace: Path):
         self._extension_map = extension_map
+        self._command_flags_map = command_flags_map
+        self._state = state
+        self._workspace = workspace
 
-    def visit(self, candidate: Path):
+    def visit(self, candidate: Path) -> List[Path]:
+        new_candidates: List[Path] = []
         try:
-            analyser = self._extension_map[candidate.suffix]
+            task_class = self._extension_map[candidate.suffix]
             reader: TextReader = FileTextReader(candidate)
             hasher: TextReaderAdler32 = TextReaderAdler32(reader)
-            analyser.analyse(hasher)
+
+            if issubclass(task_class, Analyser):
+                task: Task = task_class(hasher, self._state)
+            elif issubclass(task_class, Command):
+                flags = self._command_flags_map.get(task_class, [])
+                task = CommandTask(
+                    task_class(Path(hasher.filename), self._workspace, flags))
+            else:
+                message = 'Unhandled class "{cls}" in extension map.'
+                raise TypeError(
+                    message.format(cls=task_class))
+            # TODO: Eventually add to the queue here rather than running
+            new_candidates = task.run()
             for _ in hasher.line_by_line():
                 pass  # Make sure we've read the whole file.
-            file_info = FileInfoDatabase(analyser.database)
+            file_info = FileInfoDatabase(self._state)
             file_info.add_file_info(candidate, hasher.hash)
         except KeyError:
             pass
+        return new_candidates
 
 
 class TreeDescent(object):
     def __init__(self, root: Path):
         self._root = root
 
     def descend(self, visitor: TreeVisitor):
-        visit = [self._root]
-        while len(visit) > 0:
-            candidate: Path = visit.pop()
+        addresses = [self._root]
```

Comment:
Still doesn't seem like the best name but better than `visit`.

Reply:
I'm open to suggestions!  You can `visit` and `address` right?  (though I guess address is often used in networking and such)

Reply:
I went to networks when I saw "address". How about something simple like `to_visit` or maybe `agenda`?

Reply:
Yeah okay, I'll go with `to_visit`